app_package/src/vaporwearExperience.ts

```diff
@@ -0,0 +1,124 @@
+import "@babylonjs/loaders";
+import "@babylonjs/gui";
+
+import { Engine, SceneLoader } from "@babylonjs/core";
+import { Showroom, ShowroomState } from "./showroom";
+import { IVaporwearExperienceParams } from "./iVaporwearExperienceParams";
+import { GLTFFileLoader } from "@babylonjs/loaders";
+
+export interface IHotspotState {
+    x: number;
+    y: number;
+    isVisible: boolean;
+}
+
+export class VaporwearExperience {
+    private _showroom: Showroom;
+
+    private constructor(showroom: Showroom) {
+        this._showroom = showroom;
+    }
+
+    public static async CreateAsync(params: IVaporwearExperienceParams): Promise<VaporwearExperience> {
+        SceneLoader.OnPluginActivatedObservable.add((plugin) => {
+            if (plugin.name === "gltf") {
+                const loader = plugin as GLTFFileLoader;
+                loader.transparencyAsCoverage = true;
+            }
+        });
+
+        const canvas = params.canvas;
+        const engine = new Engine(canvas, undefined, {
+            forceSRGBBufferSupportState: true
+        });
+
+        window.addEventListener("resize", () => {
+            engine.resize();
+        });
+
+        const showroom = await Showroom.CreateAsync(engine, params);
+
+        engine.runRenderLoop(() => {
+            showroom.render();
+        });
+
+        return new VaporwearExperience(showroom);
+    }
+
+    public setZoom(zoomPercent: number): void {
+        this._showroom.setZoomPercent(zoomPercent);
+    }
+
+    public disableMouseWheel(): void {
+        this._showroom.disableMouseWheel();
+    }
+
+    public setCameraBehavior(behavior: string) {
+        switch (behavior) {
+            case "overall":
```

Comment:
Did you know you can do this to get the enum string?

```typescript
ShowroomState[ShowroomState.Overall]
```

https://www.typescriptlang.org/play?&msclkid=89e88f80bc3e11eca18539d171cbdfb4#code/KYOwrgtgBAygFgewO4CcEIjALgQy8KAbwFgAoKCqAeQDdgUcAbRgGjMqgGFGcBnABzblKAMRwBjYEI4AZYDQCWufNMqcEIAGYKA5mBTAyAXzJkm9LAAp4yNBmx5gAbRup0mZcAB0tek0YAugCUANxAA

Reply:
I did not know that, but in this case I don't want the names of the enum elements to be the same as the exposed names. Capitalization is one reason, but more directly the enum itself is an implementation detail, not part of the contract.